pandas/core/arraylike.py

```diff
@@ -296,7 +297,8 @@ def array_ufunc(self, ufunc: np.ufunc, method: str, *inputs: Any, **kwargs: Any)
         # so this ends up just being x1.index | x2.index, but we write
         # it to handle *args.
 
-        if len(set(types)) > 1:
+        set_types = set(types)
+        if len(set_types) > 1 and type(DataFrame) in set_types:
```

Comment:
What would happen if s would be a DataFrame instead of a Series? Not really familiar here, but shouldn't we rather check if Series *and* DataFrame are in the types and raise in that case?

Reply:
My mistake. Originally, the I should write like this. 
```
if len(set_types) > 1 and DataFrame in set_types:
```
What would happen if s would be a DataFrame instead of a Series?
It will failed. 
```
def add3(x, y, z):
    return x + y + z

ufunc = np.frompyfunc(add3, 3, 1)
df = pd.DataFrame(np.array([1, 2]))

print(ufunc(df, df, 1))
```
```
NotImplementedError: Cannot apply ufunc <ufunc 'add3 (vectorized)'> to mixed DataFrame and Series inputs.
```

Reply:
but shouldn't we rather check if Series and DataFrame are in the types and raise in that case?
Yes. It should be like this. I will update code and update test case.